File: M3Tools/Controls/PriceInput.vb
Public Class PriceInput

	<ComponentModel.DefaultValue("Price")>
	Public Property Label As String
		Get
			Return lbl_Price.Text
		End Get
		Set(value As String)
			lbl_Price.Text = value
		End Set
	End Property

	<ComponentModel.DefaultValue(0)>
	Public Property Price As Decimal
		Get
			Return CDec(txt_Price.Text)
		End Get
		Set(value As Decimal)
			txt_Price.Text = value.FormatPrice
		End Set
	End Property

	Private Sub FormatText() Handles txt_Price.LostFocus
		txt_Price.Text = Price.FormatPrice
	End Sub
End Class

File: M3Tools/Controls/PriceInput.cs
using System;
using Microsoft.VisualBasic.CompilerServices;
using SPPBC.M3Tools.Types.Extensions;

namespace SPPBC.M3Tools
{
    public partial class PriceInput
    {

		/// <summary>
		/// What the label should say
		/// </summary>
        [System.ComponentModel.DefaultValue("Price")]
        public string Label
        {
            get
            {
                return lbl_Price.Text;
            }
            set
            {
                lbl_Price.Text = value;
            }
        }

		/// <summary>
		/// The price to display
		/// </summary>
        [System.ComponentModel.DefaultValue(0)]
        public decimal Price
        {
            get
            {
                return Conversions.ToDecimal(txt_Price.Text);
            }
            set
            {
                txt_Price.Text = value.FormatPrice();
            }
        }

		/// <summary>
		/// 
		/// </summary>
        public PriceInput()
        {
            InitializeComponent();
        }

        private void FormatText()
        {
            txt_Price.Text = Price.FormatPrice();
        }

        private void FormatText(object sender, EventArgs e) => FormatText();
    }
}
